tensorflow/core/platform/file_system.cc

```diff
@@ -259,4 +261,50 @@ Status FileSystem::RecursivelyCreateDir(const string& dirname) {
   return Status::OK();
 }
 
+Status FileSystem::CopyFile(const string& src, const string& target) {
+  uint64 size;
+  Status s = GetFileSize(src, &size);
+  if (!s.ok()) {
+    return s;
+  }
+  std::unique_ptr<RandomAccessFile> src_file;
+  s = NewRandomAccessFile(src, &src_file);
+  if (!s.ok()) {
+    return s;
+  }
+  std::unique_ptr<WritableFile> target_file;
+  s = NewWritableFile(target, &target_file);
+  if (!s.ok()) {
+    return s;
+  }
+
+  uint64 offset = 0;
+  while (offset < size) {
+    char scratch[kCopyFileBufferSize];
+    StringPiece result;
+    size_t chunk =
+        offset + sizeof(scratch) < size ? sizeof(scratch) : size - offset;
+    s = src_file->Read(offset, chunk, &result, scratch);
+    if (!s.ok()) {
+      return s;
+    }
+
+    s = target_file->Append(result);
+    if (!s.ok()) {
+      return s;
+    }
+
+    offset += chunk;
+  }
+  s = GetFileSize(src, &size);
+  if (!s.ok()) {
+    return s;
+  }
+  if (offset != size) {
+    return errors::Aborted("File ", src, " changed while reading: ", size,
```

Comment:
In this case, we'll end up copying something into target_file. Do those semantics make sense? Or should we delete target_file? What are the semantics in Unix for instance? Either way, can we document this in the header file?

Reply:
@rohan100jain That was to follow the old semantics (where no partial copy in memory). Now as we use streaming, it is not possible to rollback any way. The `cp` in Linux is not impacted by the file change so I just removed the check at the end.